docs/DeveloperGuide.md

```diff
@@ -4,35 +4,261 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+### Architecture Diagram
+![Architecture Diagram](images/ArchitectureDiagram.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+*Figure 1: Architecture Diagram*
 
+The Architecture Diagram given above explains the high-level design of the App.
+Listed below is a brief summary outlining the primary components and their interrelationships.
 
-## Product scope
-### Target user profile
+In summary, the user interacts with the UI components, initiating a sequence that involves:
+- Parser component for command interpretation
+- Command component for execution 
+- Data component for managing application data
 
-{Describe the target user profile}
+The Storage component mainly handles interaction with external text files and main coordinates the interactions between the various Components.
 
-### Value proposition
+The bulk of the app’s work is done by the following components:
+- `UI` : The UI of the App.
+- `Storage` : Reads data from, and writes data to, the text files.
+- `Data` : Consists of all the classes that are involved in execution of commands.
+- `Parser` : Makes sense of user input to return the appropriate command
+- `Command` : Executes the comman requested by the user.
+
+How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+
+![Architecture Encode Data](images/sequence/Architecture_Encode_Data.png)
+*Figure 2: Architecture Encode Sequence Diagram*
+
+### Ui Component
+API: [Ui.java]({repoURL}src/main/java/seedu/cafectrl/ui/Ui.java)
+
+![Ui Class Diagram](images/class/Ui.png)
+<br>*Figure 3: Ui Class Diagram*
+
+The `Ui` component is responsible for interacting with the user. Within CafeCtrl, `Ui` is instantiated by `Parser`, `Command`, `Main`, `Data`, and `Storage` components to access the print methods in `Ui.java`.
+
+In the Ui component,
+- `Ui.java` consists of multiple methods that received the user input and prints messages to the system console for users to see
+- `Messages.java` consists of multiple strings that contains greeting, command, and goodbye messages to be shown to user
+- `ErrorMessages.java` consists of multiple strings that contain error messages to be shown to user when an incorrect command or exception has been returned
+
+### Parser Component
+API: [Parser.java]({repoURL}src/main/java/seedu/cafectrl/parser/Parser.java)
+
+![Parser Class Diagram](images/class/Parser.png)
+
+*Figure 4: Parser Class Diagram*
```

Comment:
Can you make the diagram higher resolution?

It is hard to read on the website, see 
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/49584216/25351378-258a-4fce-af99-ff06c73c6207)

Reply:
Thanks for pointing out! Will edit it